packer/plugin.go

```diff
@@ -248,6 +269,13 @@ func (c *PluginConfig) discoverSingle(glob string) (map[string]string, error) {
 
 		// Look for foo-bar-baz. The plugin name is "baz"
 		pluginName := file[len(prefix):]
+
+		// if Plugin name has OS_ARCH in it, split at _v(0-9) with regex
+		if strings.Contains(pluginName, OS_ARCH) {
+			regex := regexp.MustCompile(`_v[0-9]`)
+			pluginName = regex.Split(pluginName, -1)[0]
```

Comment:
In thinking about it if the name contains OS_ARCH can we just try to grab the file part after splitting on `"_"`? 

```
pluginName := strings.SplitN(s, "_", 2)[0]
```

Reply:
Yes that's a lot cleaner! Thanks for the suggestion I'll make that change